compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -251,6 +252,27 @@ static std::string translateModuleToISA(llvm::Module &module,
   return targetISA;
 }
 
+static void checkRegisterSpilling(OpBuilder &builder, StringRef obj) {
+  uint16_t abi_version;
+  llvm::StringMap<llvm::offloading::amdgpu::AMDGPUKernelMetaData> infoMap;
+
+  if (!llvm::offloading::amdgpu::getAMDGPUMetaDataFromImage(
+          llvm::MemoryBufferRef(obj, ""), infoMap, abi_version)) {
+    for (llvm::StringMapEntry<llvm::offloading::amdgpu::AMDGPUKernelMetaData>
+             &entry : infoMap) {
```

Comment:
This type supports structured bindings: https://github.com/llvm/llvm-project/blob/3f3f7d1fd99eea891ddd643201617d22c634bbfb/llvm/include/llvm/ADT/StringMapEntry.h#L149
```suggestion
    for (const auto &[kernelName, metadata] : infoMap) {
```